packages/wallet-protocols/src/protocols/create-and-direct-fund/protocol.ts

```diff
@@ -0,0 +1,120 @@
+import { Machine, MachineConfig, sendParent } from 'xstate';
+
+import { JsonRpcCreateChannelParams } from '../../json-rpc';
+import { MachineFactory } from '../../machine-utils';
+import { Store, success } from '../..';
+
+import { ethAllocationOutcome } from '../../calculations';
+
+import { AdvanceChannel, DirectFunding } from '..';
+
+const PROTOCOL = 'create-and-direct-fund';
+
+export enum Indices {
+  Left = 0,
+  Right = 0,
+}
+
+export type Init = JsonRpcCreateChannelParams & {
+  channelId: string;
+  challengeDuration: number;
+  index: Indices;
+};
+
+export type FirstStateConstructed = Init & { type: 'FIRST_STATE_CONSTRUCTED' };
+
+export const advanceChannelArgs = (i: 1 | 3) => ({ channelId }: Init): AdvanceChannel.Init => ({
+  channelId,
+  targetTurnNum: i,
+});
+
+const constructFirstState = {
+  invoke: {
+    id: 'constructFirstState',
+    src: 'constructFirstState',
+    data: (ctx: Init) => ctx,
```

Comment:
```suggestion
```

You get access to the context and the most recent event when invoke a promise. See the example [here](https://xstate.js.org/docs/guides/communication.html#invoking-promises)